prisma/factories/plus_status.ts

```diff
@@ -0,0 +1,22 @@
+import { Factory } from "fishery";
+import { PlusStatus, PlusRegion } from "@prisma/client";
+import prisma from "../client";
+import userFactory from './user';
+import _ from "lodash";
+
+export default Factory.define<PlusStatus>(({ params, onCreate }) => {
+  onCreate(plusStatus => {
+    return prisma.plusStatus.create({ data: plusStatus });
+  });
+
+  return {
+    userId: userFactory.build().id,
+    membershipTier: 1,
+    region: _.sample([PlusRegion.EU, PlusRegion.NA]) || PlusRegion.NA,
```

Comment:
I thought `_.sample([PlusRegion.EU, PlusRegion.NA])` would be enough here, but I guess it can technically return `undefined` as well? There might be a better way to make TypeScript happy here.

That said, this is a good example of how to make a factory more dynamic and random (which can be useful for catching edge cases in your automated tests!)

Reply:
I think the method might be mistyped actually. Reading through the [docs](https://lodash.com/docs/#sample) there should be no way it returns `undefined` or `null` when given an array even if the types say so.

So in this case I think in this case the most TypeScript friendly way would actually be coercing the value: 
```suggestion
    region: _.sample([PlusRegion.EU, PlusRegion.NA])!,
```

Using the `||` operator kind of means to me it can sometime be falsy and we are preparing for it while actually we just have more knowledge than TypeScript about this case so coercing it is more clear.

As a side note if we were preparing for a falsy value here the modern JS way of doing it would be via the [?? operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator). It's better than using `||` because it only gives the right-hand value if `undefined` or `null` and not for example with `0` that's also a falsy value in JS.